### PR TITLE
Add support for ARMv8-R AArch32 FVP platform

### PR DESCRIPTION
**Pull request by julien-massot**
This PR helps to test the Cortex-r52 support provided by #42220.

The fvp_baser_armv8r_aarch32 board is mostly a copy of the
fvp_baser_armv8r present in arm64 bu select the CORTEX_R52 cpu.

Documentation has not been imported since it's the same than fvp_baser_armv8r,
also we rely on the arm64/fvp-aemv8r/fvp-aemv8r.dtsi which can be odd for an arm32
based board.
Tested with FVP_Base_AEMv8R_11.17_21

*** Booting Zephyr OS build zephyr-v3.0.0-644-g08e4612ae972  ***
Hello World! fvp_baser_aemv8r_aarch32

### CODE REVIEW

**Comment by julien-massot** [comment]
looks like there is one missing ifdef in irq_manage

**Comment by julien-massot** [comment]
> looks like there is one missing ifdef in irq_manage

solved

File: arch/arm/core/aarch32/cortex_a_r/reset.S [hunk 1]
```diff
@@ -46,7 +46,40 @@ GTEXT(z_arm_platform_init)
...
+    mcr p15, 4, r0, c1, c0, 1
+
+    /* Change EL1 exception base address */
+    ldr r0,=_vector_table
+    mcr p15, 0, r0, c12, c0, 0 /* Write to VBAR */
+
+    /* Go to SVC mode */
+    mrs r0, cpsr
+    mov r1, #MODE_SVC
+    bfi r0, r1, #0, #5
```
Comment (bbolen) [inline arch/arm/core/aarch32/cortex_a_r/reset.S:71]: A macro for "5" might make this more readable.  Another possibility to make this more readable could be
```suggestion
    bic r0, #MODE_MASK
    orr r0, #MODE_SVC
```

**Review by bbolen — CHANGES_REQUESTED** [review]

**Review by povergoing — CHANGES_REQUESTED** [review]
Would you like to provide the documents for the `fvp_baser_aemv8r_aarch32` please?
refs: 
1. https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm64/fvp_baser_aemv8r/doc
2. https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm64/fvp_base_revc_2xaemv8a/doc

File: arch/arm/core/aarch32/cortex_a_r/reset.S (hunk 1 above)
Comment (julien-massot) [reply arch/arm/core/aarch32/cortex_a_r/reset.S]: Thanks picked up your suggestion.

**Review by bbolen — APPROVED** [review]

**Comment by povergoing** [comment]
> > Would you like to provide the documents for the `fvp_baser_aemv8r_aarch32` please? refs:
> > 
> > 1. https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm64/fvp_baser_aemv8r/doc
> > 2. https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm64/fvp_base_revc_2xaemv8a/doc
> 
> I just add the documents for the `fvp_baser_aemv8r_aarch32`, it is mostly based on the two documents you mentioned. I didn't test the debugging method through ARM DS, but it used to work when I was using my trial license.

No problem, no need to describe ARM DS.

**Review by povergoing — APPROVED** [review]

**Pull request merged**